File: rb/exo_11.rb
print "En quelle année es-tu né(e) ?\n> "
year = gets.chomp.to_i
x = 0
delta = Time.new.year - year
if year.zero?
  abort("Et ton année de naissance ?\n")

else if year == Time.new.year
  abort("Tu es trop jeune !\n")
end
end

while year <= Time.new.year
  puts("Il y a #{delta} ans, tu avais #{x} ans")
  year += 1
  x += 1
  delta -= 1
end

File: rs/src/bin/exo_11.rs
use chrono::prelude::*;
use std::io;

fn main() {
    println!("En quelle année es-tu né(e) ?");
    let mut birth_buf = String::new();

    io::stdin().read_line(&mut birth_buf).unwrap();

    let current_year = Utc::today().year();
    let birth_year = birth_buf.trim().parse::<i32>().unwrap();

    if birth_year >= current_year {
        panic!("Tu ne peux pas être aussi jeune ?")
    }

    for i in birth_year..=current_year {
        println!(
            "Il y a {} ans, tu avais {} ans.",
            current_year - i,
            i - birth_year
        )
    }
}
